src/compiler/utilities.ts

```diff
@@ -3220,18 +3221,24 @@ namespace ts {
     }
 
     /**
-     * Strip off existed single quotes or double quotes from a given string
+     * Strip off existed surrounding single quotes, double quotes, or backticks from a given string
      *
      * @return non-quoted string
      */
     export function stripQuotes(name: string) {
         const length = name.length;
-        if (length >= 2 && name.charCodeAt(0) === name.charCodeAt(length - 1) && startsWithQuote(name)) {
+        if (length >= 2 && name.charCodeAt(0) === name.charCodeAt(length - 1) && isQuoteOrBacktick(name.charCodeAt(0))) {
             return name.substring(1, length - 1);
         }
         return name;
     }
 
+    function isQuoteOrBacktick(charCode: number) {
+        return charCode === CharacterCodes.singleQuote ||
+            charCode === CharacterCodes.doubleQuote ||
+            charCode === CharacterCodes.backtick;
+    }
+
     export function startsWithQuote(name: string): boolean {

```

Comment:
This is only used in completions.ts now, so we should decide whether that code should handle backticks too. I think it shouldn't -- it is filtering out module declarations from completion lists, and we don't allow the syntax

```ts
declare module `foonly` {
   ...
```

In that case, I think you should delete `startsWithQuote` and inline its code into completions.ts.

Reply:
Huh, I’m surprised to learn that’s the only usage of `startsWithQuote`.

Reply:
Thanks! I wasn't sure that I should remove exported function.

> we don't allow the syntax
> ```
> declare module `foonly` {
> ```


That was plan for next investigation/issue/PR because there are several requests for similar changes in type positions: #31697 about this case and #29331